### ISSUE TITLE
Not able to cast restraints using get_from when the restraint was included in a RestraintSet

### ISSUE THREAD

**Issue by Pellarin** [salilab/imp#926]
Refers to #925 and 6bd476de3f41cc9c45c16dffe94f112b2fb75327

I wrote a new base class for isd.Restraint that includes get_probabilty(). Committed a test `test_ISDRestraint.py` that is failing when the restraint is taken from a restraint set:

```
#!/usr/bin/env python

# imp general
import IMP
import IMP.core
import IMP.isd

# unit testing framework
import IMP.test

class ISDRestraint(IMP.isd.Restraint):
    def __init__(self,m):
        IMP.isd.Restraint.__init__(self, m, "ISDRestraint %1%")

class TestISDRestraint(IMP.test.TestCase):

    def test_setup(self):
        m=IMP.Model()
        isdr=ISDRestraint(m)       
        self.assertTrue('get_probability' in dir(isdr))
        self.assertTrue('unprotected_evaluate' in dir(isdr))
        self.assertTrue('do_get_inputs' in dir(isdr))
        self.assertTrue(isdr.unprotected_evaluate(None) == 0.0)

    def test_restraintset(self):
        m=IMP.Model()        
        rs=IMP.RestraintSet(m)
        rs.add_restraint(ISDRestraint(m))
        print(dir(IMP.isd.Restraint))
        for r in rs.get_restraints():
            isdr=IMP.isd.Restraint.get_from(r)
            self.assertTrue('get_probability' in dir(isdr))
        
if __name__ == '__main__':
    IMP.test.main()

```
here's the error:

```
Traceback (most recent call last):
  File "test_ISDRestraint.py", line 33, in test_restraintset
    isdr=IMP.isd.Restraint.get_from(r)
  File "/Bis/home/rpellari/imp-projects/imp-130116/imp-debug/lib/IMP/isd/__init__.py", line 623, in get_from
    return _object_cast_to_Restraint(o)
NameError: global name '_object_cast_to_Restraint' is not defined
```

**Comment by Pellarin** [comment]
Also how can we have and isd::RestraintSet in the isd.LogWrapper class? Or alternatively how can we get the get_probability() method available?

**Comment by benmwebb** [comment]
Should be resolved by b3c291ab137b163da02f3c791f1e0dbd2a06d403.

**Comment by Pellarin** [comment]
Thanks!
